src/clients/TwitterClient.cpp

```diff
@@ -0,0 +1,651 @@
+                         value.c_str(),
+                         CURLFORM_END);
+        }
+        curl_formadd(&info.post,
+                     &lastptr,
+                     CURLFORM_COPYNAME,
+                     "media",
+                     CURLFORM_BUFFER,
+                     info.outputFileName.c_str(),
+                     CURLFORM_BUFFERPTR,
+                     fileData,
+                     CURLFORM_BUFFERLENGTH,
+                     fileSize,
+                     CURLFORM_CONTENTTYPE,
+                     "video/mp4",
+                     CURLFORM_END);
+    } else {
+        oauthData = oauth(url, httpMethod, params);
+        headers = curl_slist_append(headers, "Content-Type: application/x-www-form-urlencoded");
+        std::vector<std::string> encodedParams;
+        std::ranges::transform(params, std::back_inserter(encodedParams), [](const auto& pair) {
+            LOG_INFO << pair.first << " : " << pair.second;
+            return fmt::format("{}={}", urlEncode(pair.first), urlEncode(pair.second));
+        });
+        urlEncodedData = fmt::to_string(fmt::join(encodedParams, "&"));
+    }
+    const char* oauthHeader = oauthData.c_str();
+    headers = curl_slist_append(headers, oauthHeader);
+    info.headers = headers;
+
+    curl_easy_setopt(info.easy_handle, CURLOPT_FOLLOWLOCATION, 1L);
+    curl_easy_setopt(info.easy_handle, CURLOPT_DEFAULT_PROTOCOL, "https");
+    curl_easy_setopt(info.easy_handle, CURLOPT_CUSTOMREQUEST, httpMethod.c_str());
+    curl_easy_setopt(info.easy_handle, CURLOPT_URL, url.c_str());
+    if(it != params.end() && it->second == "APPEND") {
+        curl_easy_setopt(info.easy_handle, CURLOPT_HTTPPOST, info.post);
+    } else {
+        curl_easy_setopt(info.easy_handle, CURLOPT_POSTFIELDS, urlEncodedData.c_str());
+    }
+    curl_easy_setopt(info.easy_handle, CURLOPT_HTTPHEADER, info.headers);
+    curl_easy_setopt(info.easy_handle, CURLOPT_WRITEFUNCTION, WriteCallback);
+    curl_easy_setopt(info.easy_handle, CURLOPT_WRITEDATA, &info.response);
+    curl_easy_setopt(info.easy_handle, CURLOPT_PRIVATE, &info);
+
+    CURLcode res = curl_easy_perform(info.easy_handle);
+
+    auto [responseCode, json] = processResponse(info.easy_handle, res, info.response);
+    info.responseCode = responseCode;
+    curl_easy_cleanup(info.easy_handle);
+    info.easy_handle = nullptr;
+    curl_formfree(info.post);
+    info.post = nullptr;
+    curl_slist_free_all(info.headers);
+    info.headers = nullptr;
+
+    if(json.isMember("media_id_string")) {
+        info.externalId = json["media_id_string"].asString();
+        LOG_INFO << "uploadVideo->info.externalId:" << info.externalId;
+    }
+    if(info.responseCode < 200 || info.responseCode >= 300) {
+        return false;
+    }
+    return true;
+}
+
+bool TwitterClient::transMediaFiles(std::vector<FileTransferInfo>& fileTransferInfos,
+                                    const TransferFunc& transferFunc) {
```

Comment:
Sweep: here an issue. 
 const TransferFunc& transferFunc) {
Replace this "std::function" with a template parameter.
Here an example how to fix it: 
template <std::predicate<DataPoint const&> Criterion>
void filter(DataSet& data, Criterion criterion) { // Compliant
  for (auto const& dataPoint : data) {
    if (criterion(dataPoint)) {
      data.markForRemoval(dataPoint);
    }
  }
}